agent/consul/rate/handler.go

```diff
@@ -30,14 +30,50 @@ var (
 type Mode int
 
 const (
+	// ModeDisabled causes rate limiting to be bypassed.
+	ModeDisabled Mode = iota
+
 	// ModePermissive causes the handler to log the rate-limited operation but
 	// still allow it to proceed.
-	ModePermissive Mode = iota
+	ModePermissive
 
-	// ModeEnforcing causes the handler to reject the rate-limted operation.
+	// ModeEnforcing causes the handler to reject the rate-limited operation.
 	ModeEnforcing
 )
 
+var modeToName = map[Mode]string{
+	ModeDisabled:   "disabled",
+	ModeEnforcing:  "enforcing",
+	ModePermissive: "permissive",
+}
+var modeFromName = map[string]Mode{
+	"disabled":   ModeDisabled,
+	"enforcing":  ModeEnforcing,
+	"permissive": ModePermissive,
+	// If the value is not found in the persisted config file, then use the
+	// disabled default.
+	"": ModeDisabled,
+}
```

Comment:
Here's that trick I mentioned so we don't have to maintain 2 maps 😄 

```suggestion
var modeFromName = func() map[string]Mode {
	vals := map[string]Mode{
		"": ModeDisabled,
	}
	for k, v := range modeToName {
		vals[v] = k
	}
	return vals
}()
```